### PR TITLE
ValidatePathError

### PR DESCRIPTION
**Pull request by chenlevy24**
Added sub class of Error: validatePathError and changed the validatePath function of jdkHelper and androidSdkTools to return Promise<Result<string, validatePathError>>

### CODE REVIEW

**Review by andreban — CHANGES_REQUESTED** [review]
LGTM, with one requestest. Let's move the error to `src/lib/errors/ValidatePathError.ts`. `types` is used to express external types (eg: a Web Manifest)

File: packages/core/src/lib/jdk/JdkHelper.ts [hunk 1]
```diff
@@ -90,18 +91,21 @@ export class JdkHelper {
...
       const file = await fsPromises.readFile(path.join(javaHome, 'release'), 'utf-8');
       if (file.indexOf('JAVA_VERSION="1.8') < 0) { // Checks if the jdk's version is 8 as needed
-        return Result.error(new Error('jdkIsNotSupported'));
+        return Result.error(new ValidatePathError('The given jdk isn\'t supported.'
+            , 'PathIsNotSupported'));
       }
     } catch (e) {
-      return Result.error(new Error(e.message));
+      return Result.error(new ValidatePathError('Did not find the release file of the jdk.',
+          'PathIsNotCorrect'));
```
Comment (PEConn) [inline packages/core/src/lib/jdk/JdkHelper.ts:108]: Could this and PathIsNotSupported be constants in `ValidatePathError`? That way when we're checking the return value of this function, we can use those constants.

Comment (chenlevy24) [reply packages/core/src/lib/jdk/JdkHelper.ts:108]: Do you mean to add those constants in addition to the ErrorCode type which we talked about?

Comment (PEConn) [reply packages/core/src/lib/jdk/JdkHelper.ts:108]: Oh, they already are a type (I'd missed `ErrorCode` and thought you were just using these as strings).

Nevermind, code looks good to me.

**Review by PEConn — APPROVED** [review]

**Review by andreban — APPROVED** [review]
LGTM, Thank you!

**Pull request merged**